internal/http/healthcheck_server_test.go

```diff
@@ -0,0 +1,166 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestHealthCheckHandler(t *testing.T) {
+	tests := []struct {
+		name           string
+		method         string
+		expectedStatus int
+		wantBody       bool
+	}{
+		{
+			name:           "GET request returns 200 and status ok",
+			method:         http.MethodGet,
+			expectedStatus: http.StatusOK,
+			wantBody:       true,
+		},
+		{
+			name:           "POST request returns 405 and status not allowed",
+			method:         http.MethodPost,
+			expectedStatus: http.StatusMethodNotAllowed,
+			wantBody:       false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := httptest.NewRequest(tt.method, "/healthz", nil)
+			w := httptest.NewRecorder()
+
+			handleHealthCheck(w, req)
+
+			if got := w.Code; got != tt.expectedStatus {
+				t.Errorf("handleHealthCheck() status = %v, want %v", got, tt.expectedStatus)
+			}
+
+			if tt.wantBody {
+				var response struct {
+					Status string `json:"status"`
+				}
+
+				if err := json.NewDecoder(w.Body).Decode(&response); err != nil {
+					t.Errorf("failed to decode response body: %v", err)
+				}
```

Comment:
And this would be

```suggestion
				err := json.NewDecoder(w.Body).Decode(&response); 
				assert.NoError(t, err)
```

Reply:
https://github.com/n8n-io/task-runner-launcher/pull/37#discussion_r1862107119